src/select/multiselect/MultiSelect.js

```diff
@@ -36,8 +36,10 @@ class MultiSelect extends Component {
       options,
       getSelectValue,
       groupValueKey,
-      sort } = this.props
-    const getVal = val => (getSelectValue({ options, valueKey, value: val }))
+      sort,
+      unfilteredOptions } = this.props
+    const opts = unfilteredOptions.length > 0 ? unfilteredOptions : options
+    const getVal = val => (getSelectValue({ options: opts, valueKey, value: val }))
```

Comment:
In `FilterMultiSelect`, we can avoid this change if we use `const { options } = this.props` directly in its custom `getSelectValue`. This would make sense to me because `MultiSelect` is supposed to be "dumber" and not know about any filtering.